okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -17,45 +17,45 @@
 
 import java.io.IOException;
 import java.io.InterruptedIOException;
+import java.util.concurrent.TimeUnit;
 
 import static okio.Util.checkOffsetAndCount;
 
 /**
- * This timeout uses a background thread to take action exactly when the timeout
- * occurs. Use this to implement timeouts where they aren't supported natively,
- * such as to sockets that are blocked on writing.
+ * This timeout uses a background thread to take action exactly when the timeout occurs. Use this to
+ * implement timeouts where they aren't supported natively, such as to sockets that are blocked on
+ * writing.
  *
- * <p>Subclasses should override {@link #timedOut} to take action when a timeout
- * occurs. This method will be invoked by the shared watchdog thread so it
- * should not do any long-running operations. Otherwise we risk starving other
- * timeouts from being triggered.
+ * <p>Subclasses should override {@link #timedOut} to take action when a timeout occurs. This method
+ * will be invoked by the shared watchdog thread so it should not do any long-running operations.
+ * Otherwise we risk starving other timeouts from being triggered.
  *
- * <p>Use {@link #sink} and {@link #source} to apply this timeout to a stream.
- * The returned value will apply the timeout to each operation on the wrapped
- * stream.
+ * <p>Use {@link #sink} and {@link #source} to apply this timeout to a stream. The returned value
+ * will apply the timeout to each operation on the wrapped stream.
  *
- * <p>Callers should call {@link #enter} before doing work that is subject to
- * timeouts, and {@link #exit} afterwards. The return value of {@link #exit}
- * indicates whether a timeout was triggered. Note that the call to {@link
- * #timedOut} is asynchronous, and may be called after {@link #exit}.
+ * <p>Callers should call {@link #enter} before doing work that is subject to timeouts, and {@link
+ * #exit} afterwards. The return value of {@link #exit} indicates whether a timeout was triggered.
+ * Note that the call to {@link #timedOut} is asynchronous, and may be called after {@link #exit}.
  */
 public class AsyncTimeout extends Timeout {
   /**
-   * Don't write more than 64 KiB of data at a time, give or take a segment.
-   * Otherwise slow connections may suffer timeouts even when they're making
-   * (slow) progress. Without this, writing a single 1 MiB buffer may never
-   * succeed on a sufficiently slow connection.
+   * Don't write more than 64 KiB of data at a time, give or take a segment. Otherwise slow
+   * connections may suffer timeouts even when they're making (slow) progress. Without this, writing
+   * a single 1 MiB buffer may never succeed on a sufficiently slow connection.
    */
   private static final int TIMEOUT_WRITE_SIZE = 64 * 1024;
 
+  /** Duration for the watchdog thread to be idle before it shuts itself down. */
+  private static final long IDLE_TIMEOUT_MILLIS = TimeUnit.SECONDS.toMillis(60);
```

Comment:
The PR has no new test cases, which is sad. I ran this with various small values and confirmed that it comes back to life okay. Not sure what else to do that isn’t terrible.